vulnerabilities/reentrancy.md

```diff
@@ -85,6 +85,7 @@ function withdraw(uint256 amount) external {
     // Transfer funds to Contract B
     (bool success,) = address(contractB).call{value: amount}("");
     require(success, "Transfer failed.");
+    balances[msg.sender] = 0;
```

Comment:
I totally got your concern but that line of code is just for the representation of the state update after the external call, this can be replaced by any piece of code, for taking any other the code became quite longer as an example.

So, I just took it as a case where this was to be done after the amount transfer but the hacker can call the contract B itslef to drain funds repeatedly